presto-hive/src/main/java/com/facebook/presto/hive/HiveColumnHandle.java

```diff
@@ -227,6 +230,12 @@ public String toString()
         return name + ":" + hiveType + ":" + hiveColumnIndex + ":" + columnType + ":" + requiredSubfields;
     }
 
+    public Column toColumn(ConnectorSession session, ColumnConverterProvider columnConverterProvider)
+    {
+        ColumnConverter columnConverter = isUserDefinedTypeEncodingEnabled(session) ? columnConverterProvider.getColumnConverter() : HiveColumnConverterProvider.DEFAULT_COLUMN_CONVERTER;
+        return new Column(name, hiveType, comment, columnConverter.getTypeMetadata(hiveType, typeName));
```

Comment:
Ideally, we only check `isUserDefinedTypeEncodingEnabled(session)` at one place to centralize the control of what convert to use. We would assume the provider is already the one we want. Check my other comments in `HiveMetadata`